packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,666 @@
+          '-----------------------------------------------------------'
+        );
+        channelService.showChannelOutput();
+      }
+    }
+
+    if (test.file) {
+      vscode.window.showTextDocument(test.file).then(() => {
+        this.eventsEmitter.emit('Show Highlight', position, isRow);
+      });
+    }
+  }
+
+  public async runSingleTest(test: Test) {
+    const tmpFolder = this.getTempFolder();
+    const builder = new ReadableApexTestRunCodeActionExecutor(
+      [test.name],
+      false,
+      tmpFolder,
+      this
+    );
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new EmptyParametersGatherer(),
+      builder
+    );
+    await commandlet.run();
+  }
+
+  public updateSelection(position: number, isRow: boolean) {
+    if (isRow) {
+      const editor = vscode.window.activeTextEditor;
+      if (editor) {
+        const line = editor.document.lineAt(position - 1);
+        const startPos = new vscode.Position(
+          line.lineNumber,
+          line.firstNonWhitespaceCharacterIndex
+        );
+        editor.selection = new vscode.Selection(startPos, line.range.end);
+        editor.revealRange(new vscode.Range(startPos, line.range.end)); // Show selection hopefully
+      }
+    } else {
+      const editor = vscode.window.activeTextEditor;
+      if (editor) {
+        const pos = editor.document.positionAt(position);
+        const line = editor.document.lineAt(pos.line);
+        const startPos = new vscode.Position(
+          line.lineNumber,
+          line.firstNonWhitespaceCharacterIndex
+        );
+        editor.selection = new vscode.Selection(startPos, line.range.end);
+        editor.revealRange(new vscode.Range(startPos, line.range.end));
+      }
+    }
+  }
+
+  public getTempFolder(): string {
+    if (vscode.workspace.rootPath) {
+      let tempFolder = ospath.resolve(vscode.workspace.rootPath, 'tmp');
+      while (fs.existsSync(tempFolder)) {
+        tempFolder = tempFolder + crypto.randomBytes(16).toString('hex');
+      }
+      return tempFolder;
+    }
+    return '';
+  }
+
+  public async runApexTests(): Promise<void> {
+    await this.refresh();
+    const tmpFolder = this.getTempFolder();
+    const builder = new ReadableApexTestRunCodeActionExecutor(
+      Array.from(ApexTestOutlineProvider.testStrings.values()),
+      false,
+      tmpFolder,
+      this
+    );
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new EmptyParametersGatherer(),
+      builder
+    );
+    await commandlet.run();
+  }
+
+  public readJSONFile(folderName: string) {
+    if (vscode.workspace.rootPath) {
+      const fullFolderName = ospath.resolve(
+        vscode.workspace.rootPath,
+        folderName
+      );
+      const jsonSummary = this.getJSONFileOutput(folderName);
+      this.UpdateTestsFromJSON(jsonSummary);
+      this._onDidChangeTreeData.fire();
+      this.eventsEmitter.emit('Delete Folder', fullFolderName);
```

Comment:
@jkalloor0701 - All messages should be externalized.
@ruthemmanuelle - Comments?

Reply:
this.eventsEmitter.emit is not a message, it launches the deleteFolder function.

Reply:
✅ - OK nothing to do here then.